### PR TITLE
Added `scp` transport for downloading static input

### LINKED ISSUE #1082: add `scp` as a transport method for static file URLs

### PR DESCRIPTION
**Pull request by jasonfleming**
resolves #1082 and I included some explanatory comments ... cleaned up the existing logic as well ... tested the new code with `scp` from `fortytwo` using adcirc-global-test mesh ... also successfully tested modified version with existing digitalocean cloud cdn using `curl` to confirm that it still works.

### CODE REVIEW

File: asgs_main.sh [hunk 1]
```diff
@@ -75,34 +75,65 @@ checkFileExistence()
...
+           *)
+              warn "$THIS: Unrecognized file type to download: '$FTYPE'."
+              URL="unknown"
+              ;;
+        esac
+        local downloadCMD
+        if [[ $URL =~ "http://" || $URL =~ "https://" ]]; then
+           logMessage "$THIS: The curl version is $(curl --version)"
+           downloadCMD="curl ${URL}/${FNAME}.xz --output ${FPATH}/${FNAME}.xz"
+        elif [[ $URL =~ "scp://" ]]; then
```
Comment (wwlwpd) [inline asgs_main.sh:121]: Sent you a DM about this one, `scp://` may not be what we want. And if it is this is not looking sufficient to handle it like how you mean here (via `scp`).

**Review by wwlwpd — CHANGES_REQUESTED** [review]
Minor code suggestion, but moreover we should discuss the `scp://` URI scheme.

File: asgs_main.sh [hunk 2]
```diff
@@ -75,34 +75,65 @@ checkFileExistence()
...
+              ;;
+           *)
+              warn "$THIS: Unrecognized file type to download: '$FTYPE'."
+              URL="unknown"
+              ;;
+        esac
+        local downloadCMD
+        if [[ $URL =~ "http://" || $URL =~ "https://" ]]; then
+           logMessage "$THIS: The curl version is $(curl --version)"
+           downloadCMD="curl ${URL}/${FNAME}.xz --output ${FPATH}/${FNAME}.xz"
```
Comment (wwlwpd) [inline asgs_main.sh:120]: We may want to turn off SSL host validation here or provide a way to do it (i.e., pass `curl` the `

Comment (jasonfleming) [reply asgs_main.sh]: Ok I've fixed this by adding `--insecure` thank you

**Comment by wwlwpd** [comment]
Turns out OpenSSH has supported `scp://...` since version 7.7:

File: asgs_main.sh (hunk 1 above)
Comment (wwlwpd) [reply asgs_main.sh:121]: See my latest message, `scp://` seems legit.

**Review by wwlwpd — DISMISSED** [review]
Provide this passes your testing, I approve it.

Please move this out of draft when you're ready, and I will approve it again and merge it.

**Review by wwlwpd — APPROVED** [review]
looks good, great advancement!

**Pull request closed without merging**